src/game/Tactical/Soldier_Profile_Type.h

```diff
@@ -180,177 +181,166 @@ struct MERCPROFILESTRUCT
+	INT16 sWisdomGain;
 	INT8 bWisdomDelta;
+	INT16 sExpLevelGain;
+	INT8 bExpLevelDelta;
+	INT16 sMarksmanshipGain;
 	INT8 bMarksmanshipDelta;
-	INT8 bMedicalDelta;
-	INT8 bMechanicDelta;
+	INT16 sExplosivesGain;
 	INT8 bExplosivesDelta;
-	INT8 bStrengthDelta;
-	INT8 bLeadershipDelta;
-	UINT16 usKills;
-	UINT16 usAssists;
-	UINT16 usShotsFired;
-	UINT16 usShotsHit;
-	UINT16 usBattlesFought;
-	UINT16 usTimesWounded;
-	UINT16 usTotalDaysServed;
-
-	INT16 sLeadershipGain;
-	INT16 sStrengthGain;
-
-
+	INT16 sMechanicGain;
+	INT8 bMechanicDelta;
+	INT16 sMedicalGain;
+	INT8 bMedicalDelta;
 
-	// BODY TYPE SUBSITUTIONS
-	UINT32 uiBodyTypeSubFlags;
+	UINT16 usStatChangeChances[12]; // used strictly for balancing, never shown!
+	UINT16 usStatChangeSuccesses[12]; // used strictly for balancing, never shown!
 
-	INT16 sSalary;
-	INT8 bLife;
-	INT8 bDexterity; // dexterity (hand coord) value
 	INT8 bPersonalityTrait;
 	INT8 bSkillTrait;
-
-	INT8 bReputationTolerance;
-	INT8 bExplosive;
 	INT8 bSkillTrait2;
-	INT8 bLeadership;
-
-	INT8 bBuddy[5];
-	INT8 bHated[5];
-	INT8 bExpLevel; // general experience level
-
-	INT8 bMarksmanship;
-	INT8 bWisdom;
-
-	UINT8 bInvStatus[19];
-	UINT8 bInvNumber[19];
-	UINT16 usApproachFactor[4];
-
-	INT8 bMainGunAttractiveness;
-	INT8 bAgility; // agility (speed) value
+	INT8 bAttitude;
+	UINT8 bSexist;
 
-	BOOLEAN fUseProfileInsertionInfo; // Set to various flags, ( contained in TacticalSave.h )
+	/* Contract */
+	INT8 bMercStatus; //The status of the merc. If negative, see flags at the top of this file. Positive: The number of days the merc is away for. 0: Not hired but ready to be.
+	INT8 bReputationTolerance;
+	INT8 bDeathRate;
+	UINT32 uiDayBecomesAvailable; //day the merc will be available. used with the bMercStatus
+	INT16 sSalary;
+	UINT32 uiWeeklySalary;
+	UINT32 uiBiWeeklySalary;
+	INT8 bMedicalDeposit; // Is medical deposit required?
+	UINT16 sMedicalDepositAmount;
+	INT32 iMercMercContractLength; //Used for MERC mercs, specifies how many days the merc has gone since last page
+	UINT16 usOptionalGearCost;
+	UINT8 ubSuspiciousDeath;
+	UINT8 ubDaysOfMoraleHangover; // used only when merc leaves team while having poor morale
+	/* Locations */
+	SGPSector sSector;
 	INT16 sGridNo; // The Gridno the NPC was in before leaving the sector
-	UINT8 ubQuoteActionID;
-	INT8 bMechanical;
-
-	UINT8 ubInvUndroppable;
+	INT16 sPreCombatGridNo;
+	UINT8 ubStrategicInsertionCode;
+	UINT16 usStrategicInsertionData;
+	BOOLEAN fUseProfileInsertionInfo; // Set to various flags, ( contained in TacticalSave.h )
+	INT8 bTown;
+	INT8 bTownAttachment;
 	UINT8 ubRoomRangeStart[2];
-	UINT16 inv[19];
+	UINT8 ubRoomRangeEnd[2];
 
-	UINT16 usStatChangeChances[ 12 ]; // used strictly for balancing, never shown!
-	UINT16 usStatChangeSuccesses[ 12 ]; // used strictly for balancing, never shown!
+	INT8 bBuddy[5]{ -1, -1, -1, -1, -1 }; // Only indices 0, 1, 2 are used. Contain id's for friend1, friend2 and eventual friend respectively
```

Comment:
I know this is already wrong in the current state but these are profile ID's so the type of this array and several other members of this struct should be ProfileID or at least UINT8. I'm not sure if we would run into problems if a mod added recruitable characters with profile IDs > 127 otherwise.

Reply:
We would certainly run into problems:
1. Currently only IDs < 75 (`NUM_RECRUITABLE` I had to add) can be hired. For example, in this same struct it's why `bMercOpinion[]`'s size is 75.
2. We need "-1" in these members because it's used throughout the codebase to denote a buddy or enemy slot is vacant. "0" would be BARRY's ID so we can't use it.
3. Besides, this structure is saved into a save file which is another headache.

So, I think changing the type is not realistic within this PR. It should be a separate undertaking.

Reply:
I forgot about BuddySlot and HatedSlot enums I added in #2059. I will use them as the type.